plugins/docker/docker_if_

```diff
@@ -0,0 +1,216 @@
+
+=head1 AUTHOR
+
+Based on if_ plugin from Diego Elio Pettenò <flameeyes@flameeyes.eu>
+
+=head1 LICENSE
+
+GPLv2
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf suggest
+
+=cut
+
+. $MUNIN_LIBDIR/plugins/plugin.sh
+
+CONTAINER=${0##*docker_if_}
+
+function veth_interface_for_container() {
+        local pid=$(docker inspect -f '{{.State.Pid}}' "${1}")
+        mkdir -p /var/run/netns
+        ln -sf /proc/$pid/ns/net "/var/run/netns/${1}"
+        local index=$(ip netns exec "${1}" ip link show eth0 | head -n1 | sed s/:.*//)
+        let index=index+1
+        z=`ip link show | grep "^${index}:" | sed "s/${index}: \(.*\):.*/\1/"`
+        echo $z | cut -f 1 -d '@'
+        rm -f "/var/run/netns/${1}"
+}
+
+INTERFACE=$(veth_interface_for_container ${CONTAINER})
+
+# Who whould have thought it's so much work to determine the
+# maximum speed of a network interface.  Buckle up!
+findspeed_mbps() {
+    # Do not use interface name to guess technology.  Many many
+    # wifi drivers use "eth*" names.
+    IWLIST=$(type -p iwlist)
+    if [[ -x "$IWLIST" ]]; then
+        SPEED=$($IWLIST $INTERFACE rate 2>&1 |
+            awk 'BEGIN { RATE="U" }
+                       { if ($2 == "Mb/s") RATE=$1; }
+                 END   { print RATE; }')
+
+        if [[ "$SPEED" != "U" ]]; then
+            echo $SPEED
+            return
+        fi
+    fi
+
+    # sysfs can report the speed if the driver supports it (but it
+    # doesn't work as well for wireless cards, thus why we check for
+    # iwlist first)
+    if [[ -r /sys/class/net/$INTERFACE/speed ]]; then
+            SPEED=$(cat /sys/class/net/$INTERFACE/speed 2>/dev/null)
+            if [[ "$SPEED" -gt 0 ]]; then
+                echo $SPEED
+                return
+            fi
+    fi
+
+    ETHTOOL=$(type -p ethtool)
+    if [[ -x "$ETHTOOL" ]]; then
+        SPEED=$($ETHTOOL $INTERFACE 2>&1 |
+                 awk '/Speed:/ { gsub(/[^0-9]*/,"",$2); print $2; }')
+
+        if [[ $SPEED == [0-9]* ]]; then
+            echo $SPEED
+            return
+        fi
+    fi
+
+    MIITOOL=$(type -p mii-tool)
+    if [[ -x $MIITOOL ]]; then
+        case $($MIITOOL $INTERFACE 2>&1) in
+            *1000base*) echo 1000; return ;;
+            *100base*)  echo 100; return ;;
+            *10base*)   echo 10; return ;;
+        esac
+    fi
+
+    echo U
+}
+
+findspeed() {
+    if [[ "$speed" ]]; then
+        SPEED=$speed
+    else
+        SPEED=$(findspeed_mbps)
+    fi
+
+    if [[ -z "$SPEED" ]] || [[ "$SPEED" == "U" ]]; then
+        printf "up.info Traffic of the %s interface. Unable to determine interface speed." $INTERFACE
+        if [[ $EUID -ne 0 ]]; then
+                echo " Please run the plugin as root."
+        else
+                echo " Please install ethtool, wireless-tools, mii-tool or whatever is appropriate for the interface."
+        fi
+
+        return
+    fi
+
+    BPS=$(( $SPEED * 1000 * 1000 ))
+
+    cat <<EOF
```

Comment:
I would recommend (for clarity) to rename this function (e.g. `get_interface_bandwidth`) and return only the `BPS` value. The caller should afterwards put together the "config" lines (`u.max`, `u.info` and so on).
This will be easier to maintain in the future.